scipy/special/orthogonal.pyi

```diff
@@ -0,0 +1,320 @@
+from typing import (
+    Any,
+    Callable,
+    List,
+    Optional,
+    overload,
+    Tuple,
+    Union,
+)
+from typing_extensions import Literal
+
+import numpy
+
+_IntegerType = Union[int, numpy.integer]
+_FloatingType = Union[float, numpy.floating]
+_PointsAndWeights = Tuple[numpy.ndarray, numpy.ndarray]
+_PointsAndWeightsAndMu = Tuple[numpy.ndarray, numpy.ndarray, float]
+
+__all__ = [
+    'legendre',
+    'chebyt',
+    'chebyu',
+    'chebyc',
+    'chebys',
+    'jacobi',
+    'laguerre',
+    'genlaguerre',
+    'hermite',
+    'hermitenorm',
+    'gegenbauer',
+    'sh_legendre',
+    'sh_chebyt',
+    'sh_chebyu',
+    'sh_jacobi',
+    'roots_legendre',
+    'roots_chebyt',
+    'roots_chebyu',
+    'roots_chebyc',
+    'roots_chebys',
+    'roots_jacobi',
+    'roots_laguerre',
+    'roots_genlaguerre',
+    'roots_hermite',
+    'roots_hermitenorm',
+    'roots_gegenbauer',
+    'roots_sh_legendre',
+    'roots_sh_chebyt',
+    'roots_sh_chebyu',
+    'roots_sh_jacobi',
+]
+
+@overload
```

Comment:
The verbose typing is needed to capture the fact that the `roots_*` functions return either 2 tuples or 3 tuples depending on the value of `mu`. The overloads are for

- default `mu`
- `mu = False`
- `mu = True`

You might think that we could get away with just the latter two cases, but we actually can't: if we try to do that then we have to do `mu: Literal[...] = ...`, and because of the `= ...` in the case where `mu` isn't specified Mypy has no way of determining what the right return type is.